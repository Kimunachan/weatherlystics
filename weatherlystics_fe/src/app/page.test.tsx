import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import Page from "./page";
import {
  render,
  waitFor,
  screen,
  waitForElementToBeRemoved,
} from "@/utils/customTestUtils";

const axiosMock = new MockAdapter(axios);

describe("Main Page", () => {
  it("should render the main page", async () => {
    axiosMock
      .onGet("http://worldtimeapi.org/api/timezone")
      .replyOnce(200, ["Region1"]);
    // Arrange

    const { container } = render(await Page());
    // Act

    await waitForElementToBeRemoved(() => screen.getByText("Loading..."));

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
    // Assert

    expect(screen.getByText("Weatherlystics")).toBeInTheDocument();
  });
});
