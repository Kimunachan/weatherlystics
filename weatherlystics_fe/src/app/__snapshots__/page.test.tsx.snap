// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page should render the main page 1`] = `
<div>
  <div
    class="body"
  >
    <header
      class="header"
    >
      <div
        class="header"
      >
        Weatherlystics
      </div>
    </header>
    <main
      class="main"
    >
      <div
        class="card"
      >
        <form
          class="form"
        >
          <div
            class="form-row"
          >
            <label>
              Latitude:
              <input
                name="lat"
                step="1e-7"
                type="number"
              />
            </label>
            <label>
              Longitude:
              <input
                name="long"
                step="1e-7"
                type="number"
              />
            </label>
          </div>
          <div
            class="form-row"
          >
            <label>
              Timezone:
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                  role="log"
                />
                <div
                  class=" css-er85fr-control"
                >
                  <div
                    class=" css-1fdsijx-ValueContainer"
                  >
                    <div
                      class=" css-1jqq78o-placeholder"
                      id="react-select-2-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class=" css-1ji2ryk-Input"
                      data-value=""
                    >
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-describedby="react-select-2-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-2-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-1u9des2-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-yl5mvd-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </label>
            <button
              class="normalButton"
              type="button"
            >
              Use my location
            </button>
          </div>
          <div
            class="form-row"
          >
            <label>
              Date:
              <input
                name="date"
                type="date"
                value="2024-05-10"
              />
            </label>
            <button
              class="circleButton"
            >
              add
            </button>
          </div>
          <button
            class="longButton"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <div
        class="sectionTwo"
      >
        <div
          class="sectiontwo"
        />
      </div>
      <div
        class="sectionThree"
      />
    </main>
    <footer
      class="footer"
    >
      <div
        class="footer-content"
      >
        <p>
          Â© 2024 Weatherlystics
        </p>
      </div>
    </footer>
  </div>
</div>
`;
